Peer Code Review - Guidance Note

Avnish95 opened the pull request.

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+	    
+	  
+		\setlength\itemsep{-0.1em}
+		\item Teams \href{https://survey.wb.surveycto.com/collect/code_review_sign_up?caseid=}{sign-up} to participate in the code review each quarter. All DIME Research Assistants working on code are expected to participate, and any Bank staff or consultant who would like to get feedback on their code is welcome to participate.
```
We should think about this link. as it will be public we should make sure sign-up is off except during the designated sign-up periods, or make sure that the dates for the next round are clear. like it wouldn't be a problem if someone signs up from another unit now if they know the round they are signing up for doesn't take place until February. maybe ideal is that in the periods in-between sign-ups there's just a message that says very clearly that the next round won't be until X date and that sign-up will begin at Y date.

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+		\item All participants prepare a code package to submit to review, following \href{https://github.com/worldbank/dime-standards/blob/master/dime-coding-standards/checklists/Peer%20Code%20Review%20Submission%20Checklist.pdf}{these guidelines}. Submitted code should be \textbf{no longer than 1000 lines}. 
+        
+        \begin{itemize}
+            \item As part of this step, teams should evaluate if they would like to receive comments on \textbf{code reproducibility} and \textbf{output replicability}. 
```
```suggestion
            \item As part of this step, teams will identify what tasks they want review (data cleaning, construction, or analysis) and whether they want the reviewer to assess  \textbf{computational reproducibility}. 
```

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+        
+        \begin{itemize}
+            \item As part of this step, teams should evaluate if they would like to receive comments on \textbf{code reproducibility} and \textbf{output replicability}. 
+            \item \textbf{Note} that in this case, teams should prepare a \textbf{de-identified} version of the dataset to include in the submission package.
```
```suggestion
            \item \textbf{Note} In order for the reviewer to assess reproducibility, the peer review submission package must include a \textbf{de-identified} version of the dataset.
```

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+            \item \textbf{Note} that in this case, teams should prepare a \textbf{de-identified} version of the dataset to include in the submission package.
+        \end{itemize}
+
+        \item Participants are then paired for review based on preferred statistical software and length of code.
```
```suggestion
        \item Participants are paired for review based on preferred statistical software and length of code.
```

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+        \item The actual peer code review activity takes place over the course of one week, which includes:
+         \begin{itemize}
+            \item A \textbf{kickoff session} with an overview of the process, and pair assignments. 
+            \item A \textbf{group work party} with technical support from DIME Analytics. Pairs work together, and answer questions to allow each other to review their submitted code.
```
```suggestion
            \item A \textbf{group work session} with technical support from DIME Analytics. Pairs work together, and answer questions to allow each other to review their submitted code.
```

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+            \item A \textbf{group work party} with technical support from DIME Analytics. Pairs work together, and answer questions to allow each other to review their submitted code.
+        \end{itemize}
+
+        \item Reviewers finally submit feedback using \href{https://survey.wb.surveycto.com/collect/code_review_summary?caseid=}{this form}. They also fill out and submit the following checklists:
```
```suggestion
       \item Reviewers review the code using structured checklists:
```

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+                \item \href{https://github.com/worldbank/dime-standards/blob/master/dime-coding-standards/checklists/Cleaning%20Code%20Review%20Checklist.pdf}{Cleaning Checklist}
+                \item \href{https://github.com/worldbank/dime-standards/blob/master/dime-coding-standards/checklists/Construction%20Code%20Review%20Checklist.pdf}{Construction Checklist}
+                \item \href{https://github.com/worldbank/dime-standards/blob/master/dime-coding-standards/checklists/Analysis%20Code%20Review%20Checklist.pdf}{Analysis Checklist}
+            \end{itemize}
```
INSERT:          
\item At the end of the week, reviewers submit \href{https://survey.wb.surveycto.com/collect/code_review_summary?caseid=}{a structured feedback form} and upload their completed checklists.

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+	\section{What teams receive}
+        \begin{itemize}
+ 
+        \item \textbf{TTLs} receive \href{https://github.com/worldbank/dime-standards/blob/peer-code-review/dime-coding-standards/checklists/Sample%20TTL%20Report.pdf}{individual reports} for their projects indicating adherence to best practices, and areas for improvement.
```
```suggestion
        \item \textbf{TTLs} receive \href{https://github.com/worldbank/dime-standards/blob/peer-code-review/dime-coding-standards/checklists/Sample%20TTL%20Report.pdf}{a standardized report} for their project indicating adherence to best practices and areas for improvement.
```

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+        \begin{itemize}
+ 
+        \item \textbf{TTLs} receive \href{https://github.com/worldbank/dime-standards/blob/peer-code-review/dime-coding-standards/checklists/Sample%20TTL%20Report.pdf}{individual reports} for their projects indicating adherence to best practices, and areas for improvement.
+        \item \textbf{Research assistants} receive detailed feedback in the form of the \href{https://github.com/worldbank/dime-standards/blob/master/dime-coding-standards/checklists/Reviewer%20Feedback%20Checklist.pdf}{Reviewer Feedback Checklist} submitted by their code review partners.
```
```suggestion
        \item \textbf{Research assistants} receive detailed feedback though the \href{https://github.com/worldbank/dime-standards/blob/master/dime-coding-standards/checklists/Reviewer%20Feedback%20Checklist.pdf}{Reviewer Feedback Checklist} submitted by their review partner.
```

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,127 @@
+ 
+        \item \textbf{TTLs} receive \href{https://github.com/worldbank/dime-standards/blob/peer-code-review/dime-coding-standards/checklists/Sample%20TTL%20Report.pdf}{individual reports} for their projects indicating adherence to best practices, and areas for improvement.
+        \item \textbf{Research assistants} receive detailed feedback in the form of the \href{https://github.com/worldbank/dime-standards/blob/master/dime-coding-standards/checklists/Reviewer%20Feedback%20Checklist.pdf}{Reviewer Feedback Checklist} submitted by their code review partners.
+        \item \textbf{DIME Analytics} prepares and shares a \href{https://github.com/worldbank/dime-standards/blob/peer-code-review/dime-coding-standards/checklists/Sample%20Overall%20Summary%20Report%20.pdf}{summary report}  for the completed round of peer code review with DIME management, as well as the teams who signed up for the code review round.
```
```suggestion
        \item All participating \textbf{TTLs and DIME Managers} receive a \href{https://github.com/worldbank/dime-standards/blob/peer-code-review/dime-coding-standards/checklists/Sample%20Overall%20Summary%20Report%20.pdf}{summary report} highlighting common strengths and weaknesses.
```

mariaruth reviewed (changes requested):
This is nice, it's very useful to see everything together in one place. there was also a google sheet used at some point this round, is that always used? if so should be noted in the process here. also it might be useful to add indicative timelines for outputs, like within 2 weeks teams receive ...

Avnish95 commented:
Thank you so much for all your comments on this - I've updated it now, and added timelines for outputs.

Re: the Google sheet, we use it for internal coordination and setting up pairs/tracking submission and receipt of packages - do you think it still makes sense to mention?

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,131 @@
+		\item All participants prepare a code package to submit to review, following \href{https://github.com/worldbank/dime-standards/blob/master/dime-coding-standards/checklists/Peer%20Code%20Review%20Submission%20Checklist.pdf}{these guidelines}. Submitted code should be \textbf{no longer than 1000 lines}. 
+        
+        \begin{itemize}
+            \item As part of this step, teams will identify what tasks they want review (data cleaning, construction, or analysis) and whether they want the reviewer to assess  \textbf{computational reproducibility}. 
```
```suggestion
            \item As part of this step, teams will identify what tasks they want reviewed (data cleaning, construction, or analysis) and whether they want the reviewer to assess  \textbf{computational reproducibility}. 
```

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,131 @@
+
+       \item Reviewers review the code using structured checklists:
+        \begin{itemize}
+            \item \textbf{Mandatory}: \href{https://github.com/worldbank/dime-standards/blob/master/dime-coding-standards/checklists/Reviewer%20Feedback%20Checklist.pdf}{Reviewer Feedback Checklist}
```
Given that reproducibility check isn't mandatory, i think we want to remove that from the reviewer feedback sheet and create a separate checklist (the same way we have cleaning, construction, etc).

Avnish95 replied:
That sounds good, I'll make those changes to the checklists. And I also agree on the point about re-organizing the folder structure. I'll create a new one for this.

mariaruth commented on dime-coding-standards/checklists/tex/peer_code_review_guidance_note.tex:
```diff
@@ -0,0 +1,131 @@
+        
+         \item At the end of the week, reviewers submit \href{https://survey.wb.surveycto.com/collect/code_review_summary?caseid=}{a structured feedback form} and upload their completed checklists.
+
+         \item \textbf{DIME Analytics} conducts a \textbf{code review summary session} \textbf{(within 3 weeks)} with RAs, as well as TTLs (optional), to discuss summary statistics and general lessons from the recently concluded round of peer code review.
```
```suggestion
         \item \textbf{DIME Analytics} conducts a \textbf{code review summary session} \textbf{(within 3 weeks)} with participants, to discuss summary statistics and general lessons from the recently concluded round of peer code review. TTLs are invited to attend.
```

mariaruth reviewed (changes requested):
This looks really good now. just a few small comments remain. One note: the overall folder structure needs re-organization; this should not be in the 'checklists' folder.
